ably/composite_task.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"context"
+	"log"
+	"math/rand"
+	"strconv"
+	"sync"
+
+	"github.com/ably-forks/boomer"
+)
+
+func generateShardedChannelName(testConfig TestConfig, number int) string {
+	return "sharded-test-channel-" + strconv.Itoa(number%testConfig.NumChannels)
+}
+
+var compositeUserCounter int
+var compositeUserMutex sync.Mutex
+
+func compositeTask(testConfig TestConfig) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	boomer.Events.Subscribe("boomer:stop", cancel)
+
+	errorChannel := make(chan error)
+
+	client, err := newAblyClient(testConfig)
+	defer client.Close()
+
+	if err != nil {
+		boomer.RecordFailure("ably", "subscribe", 0, err.Error())
+		return
+	}
+
+	compositeUserMutex.Lock()
+	compositeUserCounter++
+	userNumber := compositeUserCounter
+	compositeUserMutex.Unlock()
```

Comment:
You may want to use [`atomic.AddInt64`](https://golang.org/pkg/sync/atomic/#AddInt64) to do this in one step.